Fix match type not being propagated correctly to other users' settings

peppy opened the pull request.

peppy committed:
Add test coverage of match type propagating to other users' settings

peppy committed:
Fix missing transfer of match type to settings

smoogipoo commented on osu.Game.Tests/Visual/Multiplayer/TestSceneTeamVersus.cs:
```diff
@@ -118,6 +119,25 @@ public void TestChangeTeamsViaButton()
         }
 
+        [Test]
+        public void TestSettingsUpdatedWhenChangingQueueMode()
```
when changing match type*

peppy committed:
Fix incorrect test step text

Co-authored-by: Dan Balasescu <smoogipoo@smgi.me>

smoogipoo commented on osu.Game.Tests/Visual/Multiplayer/TestSceneTeamVersus.cs:
```diff
@@ -118,6 +119,25 @@ public void TestChangeTeamsViaButton()
+            createRoom(() => new Room
+            {
+                Name = { Value = "Test Room" },
+                Type = { Value = MatchType.TeamVersus }
+            });
+
+            AddUntilStep("match type versus", () => client.APIRoom?.Type.Value == MatchType.HeadToHead);
```
This assertion is weird. The room seems like it should be created in HeadToHead mode.

bdach replied:
I just went ahead and fixed this because I don't see any conceivable way that starting in head to head was not the original intention.

I've also had to specify an initial playlist item to fix headless failures, and I added an until step checking that the "create room" button is enabled to boot to make the cause of failure more obvious there (without it it just hangs waiting for a room join that will never happen).

bdach committed:
Add extra until step to make cause of potential failures more obvious

bdach committed:
Fix test failures due to wrong asserts & uninitialised playlist

* The "create room" button was disabled headless due to not specifying
  the imported beatmap. In visual tests it seems to work due to
  selecting from the local database randomly.

* The test asserts are brought in line with expectations.
